qiskit/compiler/scheduler.py

```diff
@@ -65,15 +67,30 @@ def schedule(
     Raises:
         QiskitError: If ``inst_map`` and ``meas_map`` are not passed and ``backend`` is not passed
     """
+    if backend is None:
+        deprecate_arg(
+            "backend",
+            additional_msg=("'backend' argument becomes required from Optional."),
+            since="0.25.0",
+        )
     arg_circuits_list = isinstance(circuits, list)
     start_time = time()
-    if backend and getattr(backend, "version", 0) > 1:
+    if isinstance(backend, BackendV2):
         if inst_map is None:
             inst_map = backend.instruction_schedule_map
         if meas_map is None:
             meas_map = backend.meas_map
         if dt is None:
             dt = backend.dt
+        target = Target.from_configuration(
+            basis_gates=backend.operation_names,
+            num_qubits=backend.num_qubits,
+            coupling_map=backend.coupling_map,
+            inst_map=inst_map,
+            meas_map=meas_map,
+            dt=dt,
+        )
```

Comment:
`from_configuration` requires basis_gates. 
How do we create target from configuration when the backend is None?